src/cast_core/src/actors/CrudActor.cpp

```diff
@@ -0,0 +1,1241 @@
+        }
+        _docTemplate = value_generators::makeDoc(insertDoc, rng);
+
+        // TODO: parse insert options.
+    }
+
+    mongocxx::model::write getModel() override {
+        bsoncxx::builder::stream::document document{};
+        _docTemplate->view(document);
+        return mongocxx::model::insert_one{document.extract()};
+    }
+
+    void run(mongocxx::client_session& session) override {
+        bsoncxx::builder::stream::document insertDoc{};
+        _docTemplate->view(insertDoc);
+        auto view = insertDoc.view();
+        auto ctx = _operation.start();
+        (_onSession) ? _collection.insert_one(session, view, _options)
+                     : _collection.insert_one(view, _options);
+        ctx.success();
+    }
+
+private:
+    bool _onSession;
+    mongocxx::collection _collection;
+    std::unique_ptr<DocGenerator> _docTemplate;
+    metrics::Operation _operation;
+    mongocxx::options::insert _options;
+};
+
+struct UpdateOneOperation : public WriteOperation {
+    UpdateOneOperation(YAML::Node opNode,
+                       bool onSession,
+                       mongocxx::collection collection,
+                       genny::DefaultRandom& rng,
+                       metrics::Operation operation)
+        : _onSession{onSession}, _collection{collection}, _operation{operation} {
+        auto filter = opNode["Filter"];
+        auto update = opNode["Update"];
+        if (!filter || !update) {
+            throw InvalidConfigurationException(
+                "'updateOne' expects 'Filter' and 'Update' fields.");
+        }
+        _filterTemplate = value_generators::makeDoc(filter, rng);
+        _updateTemplate = value_generators::makeDoc(update, rng);
+
+        // TODO: parse update options.
+    }
+
+    mongocxx::model::write getModel() override {
+        bsoncxx::builder::stream::document filterDocument{};
+        _filterTemplate->view(filterDocument);
+
+        bsoncxx::builder::stream::document updateDocument{};
+        _updateTemplate->view(updateDocument);
+        return mongocxx::model::update_one{filterDocument.extract(), updateDocument.extract()};
+    }
+
+    void run(mongocxx::client_session& session) override {
+        bsoncxx::builder::stream::document filterDocument{};
+        _filterTemplate->view(filterDocument);
+
+        bsoncxx::builder::stream::document updateDocument{};
+        _updateTemplate->view(updateDocument);
+
+        auto filterView = filterDocument.view();
+        auto updateView = updateDocument.view();
+        auto ctx = _operation.start();
+        (_onSession) ? _collection.update_one(session, filterView, updateView, _options)
+                     : _collection.update_one(filterView, updateView, _options);
+        ctx.success();
+    }
+
+private:
+    bool _onSession;
+    mongocxx::collection _collection;
+    std::unique_ptr<DocGenerator> _filterTemplate;
+    std::unique_ptr<DocGenerator> _updateTemplate;
```

Comment:
imho template is a much better name than value-generator. nice.

consider adding `using UDocGenerator = std::unique_ptr<DocGenerator>;`